src/Stratis.Bitcoin.Features.Interop/EthereumClient/EthereumClientBase.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            try
+            {
+                // Note: this will only return events from after the filter is created.
+                return this.transferEventHandler.GetFilterChanges(this.filterId).GetAwaiter().GetResult();
+            }
+            catch (RpcResponseException)
+            {
+                // If the filter is no longer available it may need to be re-created.
+                this.CreateTransferEventFilter();
+            }
+
+            return this.transferEventHandler.GetFilterChanges(this.filterId).GetAwaiter().GetResult();
+        }
+
+        public string GetDestinationAddress(string address)
+        {
+            return WrappedStrax.GetDestinationAddress(this.web3, this.interopSettings.WrappedStraxAddress, address).GetAwaiter().GetResult();
+        }
+
+        public BigInteger GetBlockHeight()
+        {
+            var blockNumberHandler = new EthBlockNumber(this.web3.Client);
+            HexBigInteger block = blockNumberHandler.SendRequestAsync().GetAwaiter().GetResult();
+            
+            return block.Value;
+        }
+
+        /// <inheritdoc />
+        public BigInteger SubmitTransaction(string destination, BigInteger value, string data)
+        {
+            return MultisigWallet.SubmitTransaction(this.web3, this.interopSettings.MultisigWalletAddress, destination, value, data, this.interopSettings.EthereumGas, this.interopSettings.EthereumGasPrice).GetAwaiter().GetResult();
+        }
+
+        /// <inheritdoc />
+        public string ConfirmTransaction(BigInteger transactionId)
+        {
+            return MultisigWallet.ConfirmTransaction(this.web3, this.interopSettings.MultisigWalletAddress, transactionId, this.interopSettings.EthereumGas, this.interopSettings.EthereumGasPrice).GetAwaiter().GetResult();
+        }
+
+        public BigInteger GetConfirmationCount(BigInteger transactionId)
+        {
+            return MultisigWallet.GetConfirmationCount(this.web3, this.interopSettings.MultisigWalletAddress, transactionId).GetAwaiter().GetResult();
+        }
+
+        public string EncodeMintParams(string address, BigInteger amount)
+        {
+            // TODO: Extract this directly from the ABI
+            const string MintMethod = "40c10f19";
+
+            var abiEncode = new ABIEncode();
+            string result = MintMethod + abiEncode.GetABIEncoded(new ABIValue("address", address), new ABIValue("uint256", amount)).ToHex();
+
+            return result;
+        }
+
+        public string EncodeBurnParams(BigInteger amount, string straxAddress)
+        {
+            // TODO: Extract this directly from the ABI
+            const string BurnMethod = "7641e6f3";
+
+            var abiEncode = new ABIEncode();
+            string result = BurnMethod + abiEncode.GetABIEncoded(new ABIValue("uint256", amount)).ToHex();
+
+            return result;
+        }
+
+        public Dictionary<string, string> InvokeContract(InteropRequest request)
```

Comment:
some of the methods in this class are not used, some return empty values. Those probably should be implemented or removed